### PR TITLE
refactor(tile-select): tailwindify

### PR DESCRIPTION
**Pull request by julio8a**
Supports #1500

### CODE REVIEW

File: src/components/calcite-tile-select/calcite-tile-select.scss [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+  box-border
+  cursor-pointer
+  inline-block
+  h-full
+  p-3
+  relative
+  duration-150
+  ease-in-out
+  align-top
+  z-10;
```
Comment (jrowlingson) [inline src/components/calcite-tile-select/calcite-tile-select.scss:14]: Indent et al

File: src/components/calcite-tile-select/calcite-tile-select.scss [hunk 2]
```diff
@@ -1,18 +1,19 @@
-$spacing: $baseline/2;
+$spacing: theme("spacing.3");
```
Comment (jrowlingson) [inline src/components/calcite-tile-select/calcite-tile-select.scss:1]: can we kill this var?

File: src/components/calcite-tile-select/calcite-tile-select.scss [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+  duration-150
+  ease-in-out
+  align-top
+  z-10;
   box-shadow: 0 0 0 1px var(--calcite-ui-border-2);
-  box-sizing: border-box;
-  cursor: pointer;
-  display: inline-block;
-  height: 100%;
   max-width: 300px;
```
Comment (jrowlingson) [inline src/components/calcite-tile-select/calcite-tile-select.scss:16]: `max-width: 20rem` ?

File: src/components/calcite-tile-select/calcite-tile-select.scss [hunk 4]
```diff
@@ -62,14 +62,14 @@ $spacing: $baseline/2;
...
+    @apply items-center;
   }
 }
 
 :host([input-enabled][input-alignment="start"][icon][heading][description]),
 :host([input-enabled][input-alignment="start"]:not([icon])[heading]:not([description])) {
   .root {
-    display: inline-grid;
+    @apply inline-grid;
     grid-gap: $spacing;
```
Comment (jrowlingson) [inline src/components/calcite-tile-select/calcite-tile-select.scss:73]: gap-3

File: src/components/calcite-tile-select/calcite-tile-select.scss [hunk 5]
```diff
@@ -78,14 +78,14 @@ $spacing: $baseline/2;
     order: 0;
```
Comment (jrowlingson) [inline src/components/calcite-tile-select/calcite-tile-select.scss:78]: order-none

File: src/components/calcite-tile-select/calcite-tile-select.scss [hunk 6]
```diff
@@ -62,14 +62,14 @@ $spacing: $baseline/2;
...
   }
 }
 
 :host([input-enabled][input-alignment="start"][icon][heading][description]),
 :host([input-enabled][input-alignment="start"]:not([icon])[heading]:not([description])) {
   .root {
-    display: inline-grid;
+    @apply inline-grid;
     grid-gap: $spacing;
     grid-template-columns: max-content 1fr;
```
Comment (jrowlingson) [inline src/components/calcite-tile-select/calcite-tile-select.scss:74]: Did you consider refactoring to use the `grid-col-` util?

Comment (eriklharper) [reply src/components/calcite-tile-select/calcite-tile-select.scss:74]: How _would_ you use any tailwind grid helpers for something like this?  Grid stuff is usually very specific and the generalized tailwind classes don't usually accommodate the need there.

**Comment by julio8a** [comment]
@jrowlingson, how should we proceed with this one?

File: src/components/calcite-tile-select/calcite-tile-select.scss [hunk 7]
```diff
@@ -1,18 +1,17 @@
...
-  background-color: var(--calcite-ui-foreground-1);
+  @apply bg-foreground-1
+    box-border
+    cursor-pointer
+    inline-block
+    h-full
+    p-3
+    relative
+    duration-150
+    ease-in-out
```
Comment (caripizza) [inline src/components/calcite-tile-select/calcite-tile-select.scss:10]: @julio8a the [default transition from calcite-base](https://github.com/Esri/calcite-base/blob/master/dist/_variables.scss#L2) exists [in our Tailwind config](https://github.com/Esri/calcite-components/blob/master/tailwind.config.js#L231), let's make use of it. It's applicable whenever you see `transition: $transition` in the scss file. (You can use the utility like this: `@apply transition-default`.)

**Review by caripizza — CHANGES_REQUESTED** [review]
@julio8a can you update this pr branch off the latest master? This one still has the old calcite-tile styles in place.

File: src/components/calcite-tile-select/calcite-tile-select.scss (hunk 7 above)
Comment (julio8a) [reply src/components/calcite-tile-select/calcite-tile-select.scss:10]: Could this be the reason test is failing?

Comment (caripizza) [reply src/components/calcite-tile-select/calcite-tile-select.scss:10]: No, I don't think that previous fail was related

**Comment by paulcpederson** [comment]
@julio8a I just fixed a merge conflict from the tile select pr going in, FYI

**Comment by julio8a** [comment]
Still needed

**Pull request closed without merging**